### PR TITLE
fix: update nft-claimer mint signing

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

Signature implementation for NFT-Claimer is out-of-sync with the smart contract.

## 💊 Fixes / Solution

Update the NFT-Claimer signature schema to be in sync with what the smart contract is using.

## 🚧 Changes

- Update the signature schema
- Add tests

This PR only affect NFT-Claimer behavior, and can be pulled, as the signature is not live yet, and deploying this PR will allow easier testing by the SC team

## 🛠️ Tests

- Start the app 
- Run `curl -X POST localhost:3005/api/nft-claimer/mint/sign -H "Content-Type: application/json" -d '{"id": "0x28535f56f29a9b085be88e3896da573c61095a14f092ce72afea3c83f4feefe0", "address": "0x91FD2c8d24767db4Ece7069AA27832ffaf8590f3", "salt": "123456337495"}'`
- It should output something

### CODE REVIEW

File: test/.env.test [hunk 1]
```diff
@@ -1,4 +1,7 @@
 HUB_URL=https://hub.snapshot.org
 MODERATION_LIST_PATH=test/fixtures
-NFT_CLAIMER_PRIVATE_KEY=0x0565f8f29a83e5bca89b2acbd259d222fcfd8f024e76344c17275c43198fd30c
 NETWORK=1
+NFT_CLAIMER_PRIVATE_KEY=0x0565f8f29a83e5bca89b2acbd259d222fcfd8f024e76344c17275c43198fd30c
+NFT_CLAIMER_ADDRESS=0xD60349c24dB7F1053086eF0D6364b64B1e0313f0
```
Comment (ChaituVR) [inline test/.env.test:5]:
```suggestion
```

**Comment by wa0x6e** [comment]
Your latest commits are breaking the tests

**Comment by wa0x6e** [comment]
Values on .env.test should remains as is, all tests depends on them

File: src/api.ts [hunk 2]
```diff
@@ -45,14 +45,14 @@ router.get('/moderation', async (req, res) => {
   }
 });
 
-router.post('/nft-claimer/:type(space|proposal)/sign', async (req, res) => {
+router.post('/nft-claimer/:type(deploy|mint)/sign', async (req, res) => {
```
Comment (ChaituVR) [inline src/api.ts:48]: Need to update readme for this

Comment (wa0x6e) [reply src/api.ts:48]: Will come in another pr

File: src/lib/nftClaimer.ts [hunk 3]
```diff
@@ -1,74 +1,76 @@
...
 
-const NETWORK = process.env.NETWORK || '1';
+const HUB_NETWORK = process.env.NETWORK || '1';
+const NFT_CLAIMER_NETWORK = process.env.NFT_CLAIMER_NETWORK || '1';
+
 const PRIVATE_KEY = process.env.NFT_CLAIMER_PRIVATE_KEY;
 
-if (!PRIVATE_KEY) {
-  throw new Error('NFT Claimer private key missing');
+if (!PRIVATE_KEY || !process.env.NFT_CLAIMER_VERIFYING_CONTRACT) {
```
Comment (ChaituVR) [inline src/lib/nftClaimer.ts:29]:
```suggestion
const NFT_CLAIMER_VERIFYING_CONTRACT = process.env.NFT_CLAIMER_VERIFYING_CONTRACT
if (!PRIVATE_KEY || !NFT_CLAIMER_VERIFYING_CONTRACT) {
```

File: src/lib/nftClaimer.ts [hunk 4]
```diff
@@ -79,5 +82,30 @@ export async function signValidProposal(address: string, id: string, salt: numbe
...
+}
+
+async function signValidProposal(domain: string, message: Record<string, string | number>) {
+  return splitSignature(
+    await signer._signTypedData(
+      {
+        name: domain,
+        version: '0.1',
+        chainId: NFT_CLAIMER_NETWORK,
+        verifyingContract: process.env.NFT_CLAIMER_VERIFYING_CONTRACT
```
Comment (ChaituVR) [inline src/lib/nftClaimer.ts:105]:
```suggestion
        verifyingContract: NFT_CLAIMER_VERIFYING_CONTRACT
```

**Review by ChaituVR — APPROVED** [review]
tAck

**Pull request merged**